frontend/ListAds.elm

```diff
@@ -17,160 +18,231 @@ import Removal
 getAds model =
-  let
-    queryString =
-      QueryString.empty
-        |> QueryString.add "limit" (toString limit)
-        |> QueryString.add "offset" (toString model.cursor)
-        |> QueryString.optional "domain" model.selectedDomain
-        |> QueryString.optional "position" model.selectedPosition
-        |> QueryString.optional "location" model.selectedLocation
-        |> QueryString.render
-
-    url = "/api/ilmoitukset/" ++ queryString
-    request = Http.get url (Json.list Models.Ad.adDecoder)
-  in
+    let
+        queryString =
+            QueryString.empty
+                |> QueryString.add "limit" (toString limit)
+                |> QueryString.add "offset" (toString model.cursor)
+                |> QueryString.optional "domain" model.selectedDomain
+                |> QueryString.optional "position" model.selectedPosition
+                |> QueryString.optional "location" model.selectedLocation
+                |> QueryString.optional "sorting" model.selectedSorting
+                |> QueryString.render
+
+        url =
+            "/api/ilmoitukset/" ++ queryString
+
+        request =
+            Http.get url (Json.list Models.Ad.adDecoder)
+    in
     Util.errorHandlingSend (UpdateAds model.cursor) request
 
 
 view : Maybe User -> Model -> Config.Model -> H.Html (ViewMessage Msg)
 view loggedInUserMaybe model config =
-  H.div []
-    [ H.div
-      [ A.class "container" ]
-      [ H.div
-        [ A.class "row" ]
-        [ H.div
-          [ A.class "col-sm-12" ]
-          [ H.h1
-            [ A.class "list-ads__header" ]
-            [ H.text "Selaa ilmoituksia" ]
-          ]
-        ]
-      , H.div
-        [ A.class "row list-users__filters" ]
+    H.div []
         [ H.div
-          [ A.class "col-xs-12 col-sm-4" ]
-          [ Common.select "list-users" (LocalViewMessage << ChangeDomainFilter) Domain config.domainOptions model ]
+            [ A.class "container" ]
+            [ H.div
+                [ A.class "row" ]
+                [ H.div
+                    [ A.class "col-sm-12" ]
+                    [ H.h1
+                        [ A.class "list-ads__header" ]
+                        [ H.text "Selaa ilmoituksia" ]
+                    ]
+                ]
+            , H.div
+                [ A.class "row list-users__filters" ]
+                [ H.div
+                    [ A.class "col-xs-12 col-sm-3" ]
+                    [ Common.select "list-users" (LocalViewMessage << ChangeDomainFilter) Domain config.domainOptions model ]
+                , H.div
+                    [ A.class "col-xs-12 col-sm-3" ]
+                    [ Common.select "list-users" (LocalViewMessage << ChangePositionFilter) Position config.positionOptions model ]
+                , H.div
+                    [ A.class "col-xs-12 col-sm-3" ]
+                    [ Common.select "list-users" (LocalViewMessage << ChangeLocationFilter) Location Config.finnishRegions model ]
+                , H.div
```

Comment:
The sort should look the same as in `ListUsers.elm`, so not here, but as buttons above the actual list.